### ISSUE TITLE
Work with pug babel plugin?

### ISSUE THREAD

**Issue by jpap** [yannickcr/eslint-plugin-react#1168]
How might we make this plugin work with [babel-plugin-transform-react-pug](https://github.com/pugjs/babel-plugin-transform-react-pug)?

For example, ESLint reports three errors on the code below:

1. `React` is defined but never used (because it doesn't detect the JSX)
1. `onClick` is assigned a value but never used
1. `pug` is not defined

```
import React from 'react';

const Button = (props) => {
  const onClick = () => {
    props.onClick();
  };

  return pug`
    button(onClick=onClick)
      = props.label
  `;
};

export default Button;
```

**Comment by ezhlobo** [comment]
@jpap @EricBeReal hey, I started working on [eslint-plugin-react-pug](https://github.com/ezhlobo/eslint-plugin-react-pug). It's very new but it solves those issues.

**Comment by jpap** [comment]
@ezhlobo nice work! I've personally moved onto TypeScript/TSLint, but will definitely check out your ESLint plugin for reference.

**Comment by ljharb** [comment]
I'd say a separate linter plugin is the right solution here; this plugin is for react.
